homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -341,7 +344,18 @@ def __init__(
             self._attr_unique_id = f"{address}-{key}"
         if ATTR_NAME not in self._attr_device_info:
             self._attr_device_info[ATTR_NAME] = self.processor.coordinator.name
-        self._attr_name = processor.entity_names.get(entity_key)
+        if not description.translation_key or (
+            description.device_class
+            and any(
+                isinstance(description, device_class_type)
+                for device_class_type in (
+                    SensorDeviceClass,
+                    BinarySensorDeviceClass,
+                    NumberDeviceClass,
+                )
+            )
+        ):
```

Comment:
I think we could arrange for the entity_name to be None instead of checking the device class

Reply:
You mean `self.processor.coordinator.name`? Since the name in the entity description is probably undefined